### ISSUE TITLE
Auto-resolve dependencies between jobs.

### ISSUE THREAD

**Issue by topa** [epha/robot#12]
There is a two-way dependency between *swissmedic*- and *atc*-job.
*atc* requires: `data/release/swissmedic/swissmedic.json`
while 
*swissmedic* requires: `data/manual/swissmedic/atc.csv` (which is probably build during *atc*-job: `data/release/atc/atc.csv`).

Maybe parts of jobs can be split into smaller reusable sub-job that are able to satisfy dependencies.

**Comment by Jonas74** [comment]
Just make an additional job atc-ch

**Comment by topa** [comment]
With this commit https://github.com/epha/robot/commit/09f338cfbac3061e3551299db73136558064554d it is solved like that:

You choose `atc` and say `go`:

```bash
npm start
> epha-robot@0.2.39 start /Users/topa/Workspace/epha/robot
> node ./bin/cli.js

EMIL: I'm ready, if you are? Type help for help.
> atc
> go
```

First German `atc` files will be written, because the data like ```swissmedic.json``` is a dependency for the `atcCH`-job. If this has finished successfully it will start `atcCH`-job. The `atcCH`-job is a separate controller/job, but can't run stand alone (it is at the moment not choose-able in cli) due to the mentioned dependencies. However, it is able to check if there is a `swissmedic.json` present. If so it will use it, otherwise it will trigger ```swissmedic```-job automatically. This has the problem that `swissmedic.json` is potentially old, which must be checked right now manually. Though up-to-dateness can be probably assured with some code that will be implemented durin #13 Diff-Task.

```bash
> ...
>epha-robot@0.2.39 | DEBUG |  ATC Done
>epha-robot@0.2.39 |  TIME |  ATC Completed in { duration: '5638ms' }
>epha-robot@0.2.39 |  INFO |  ATC-CH Get, Load and Parse
>epha-robot@0.2.39 |  WARN |  Dependency /Users/topa/Workspace/epha/robot/data/release/swissmedic/swissmedic.json hasn't been yet resolved
>epha-robot@0.2.39 |  WARN |  Trying to auto-resolve /Users/topa/Workspace/epha/robot/data/release/swissmedic/swissmedic.json
>epha-robot@0.2.39 |  INFO |  Swissmedic Get, Load and Parse
> ...
```

Still I think there is a misunderstanding from my side. I thought `atc.csv` which is the result of the `atc`-job is a dependency for the `swissmedic`-job. But the `swissmedic`-job isn't and was never able to consume this file, right? However both files, `data/release/atc/atc.csv` and `data/manual/swissmedic/atc.csv` share the same name, they don't share the same format:

`"A","Alimentäres System und Stoffwechsel","undefined"` **VS** `16105,C05BA,C05BA01,"Hirudoid, Creme"`

**tltr**: There is no two-dependency, right?

**Comment by topa** [comment]
I think I'm right, so I'll close this one. If not just re-open it.
